be/src/exec/sorting/merge_path.cpp

```diff
@@ -0,0 +1,904 @@
+    metrics.stage_timers[detail::Stage::PROCESS] =
+            ADD_CHILD_TIMER(metrics.profile, "3-ProcessStageTime", overall_stage_timer_name);
+    metrics.stage_timers[detail::Stage::SPLIT_CHUNK] =
+            ADD_CHILD_TIMER(metrics.profile, "4-SplitChunkStageTime", overall_stage_timer_name);
+    metrics.stage_timers[detail::Stage::FETCH_CHUNK] =
+            ADD_CHILD_TIMER(metrics.profile, "5-FetchChunkStageTime", overall_stage_timer_name);
+    metrics.stage_timers[detail::Stage::PENDING] =
+            ADD_CHILD_TIMER(metrics.profile, "6-PendingStageTime", overall_stage_timer_name);
+    metrics.stage_timers[detail::Stage::FINISHED] =
+            ADD_CHILD_TIMER(metrics.profile, "7-FinishedStageTime", overall_stage_timer_name);
+
+    const std::string overall_stage_counter_name = "OverallStageCount";
+    metrics.stage_counter = ADD_COUNTER(metrics.profile, overall_stage_counter_name, TUnit::UNIT);
+
+    metrics.stage_counters.resize(detail::Stage::FINISHED + 1);
+    metrics.stage_counters[detail::Stage::INIT] =
+            ADD_CHILD_COUNTER(metrics.profile, "1-InitStageCount", TUnit::UNIT, overall_stage_counter_name);
+    metrics.stage_counters[detail::Stage::PREPARE] =
+            ADD_CHILD_COUNTER(metrics.profile, "2-PrepareStageCount", TUnit::UNIT, overall_stage_counter_name);
+    metrics.stage_counters[detail::Stage::PROCESS] =
+            ADD_CHILD_COUNTER(metrics.profile, "3-ProcessStageCount", TUnit::UNIT, overall_stage_counter_name);
+    metrics.stage_counters[detail::Stage::SPLIT_CHUNK] =
+            ADD_CHILD_COUNTER(metrics.profile, "4-SplitChunkStageCount", TUnit::UNIT, overall_stage_counter_name);
+    metrics.stage_counters[detail::Stage::FETCH_CHUNK] =
+            ADD_CHILD_COUNTER(metrics.profile, "5-FetchChunkStageCount", TUnit::UNIT, overall_stage_counter_name);
+    metrics.stage_counters[detail::Stage::PENDING] =
+            ADD_CHILD_COUNTER(metrics.profile, "6-PendingStageCount", TUnit::UNIT, overall_stage_counter_name);
+    metrics.stage_counters[detail::Stage::FINISHED] =
+            ADD_CHILD_COUNTER(metrics.profile, "7-FinishedStageCount", TUnit::UNIT, overall_stage_counter_name);
+}
+
+bool MergePathCascadeMerger::_is_current_stage_done() {
+    return std::all_of(_process_cnts.begin(), _process_cnts.begin() + _working_parallelism,
+                       [](auto& cnt) { return cnt.load(std::memory_order_relaxed) == 0; });
+}
+
+void MergePathCascadeMerger::_forward_stage(const detail::Stage& stage, int32_t worker_num,
+                                            std::vector<size_t>* process_cnts) {
+    // Enter critical section
+    _is_forwarding_stage.store(true, std::memory_order_relaxed);
+
+    _stage.store(stage, std::memory_order_relaxed);
+    _working_parallelism = worker_num;
+    for (auto& cnt : _process_cnts) {
+        cnt.store(1, std::memory_order_relaxed);
+    }
+    if (process_cnts != nullptr) {
```

Comment:
is better?
```
if (process_cnts != nullptr) {
   for (auto& cnt : _process_cnts) {
        cnt.store(1, std::memory_order_relaxed);
    }
```

Reply:
Maybe you've misunderstood. If process_cnts is not nullptr. every parallelism need to handle once. If process_cnts is not nullptr, respect it.